### PR TITLE
feat(deleteSubscription): feature/icky-832-My-Account-Delete-Subscription

### PR DESCRIPTION
**Pull request by kibo-Kripa**

### CODE REVIEW

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx [hunk 1]
```diff
@@ -147,6 +149,23 @@ const SubscriptionItem = (props: SubscriptionItemProps) => {
...
+    const params = {
+      subscriptionId: subscriptionId,
+      subscriptionItemId: subscriptionItemId,
+      subscriptionReasonInput: {
+        actionName: 'cancel',
+        reasonCode: 'cancel',
+        description: 'cancel',
+        moreInfo: 'cancel',
+      },
+    }
```
Comment (Kibo-TriptiV) [inline components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx]: Should we have separate buildSubscriptionParam file and build params for all the subscription actions?
Also please take action name from constant file.

File: hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * [Mutation hook] useDeleteSubscriptionMutation uses the graphQL mutation
+ *
+ * <b>deleteSubscription(subscriptionId: string): Subscription</b>
+ *
+ * Description : Delete subscription order according to the actionName.
+ *
+ * Parameters passed to function skipNextSubscription(subscriptionId?: string | null) => expects subscriptionId
+ *
+ * @returns 'response?.subscription' which contains next order date when the order will be placed
```
Comment (Kibo-TriptiV) [inline hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: Something wrong description written for delete subscription.
Also, graphql accepts 3 or 4 params, not only subscriptionId
deleteSubscriptionItem(subscriptionId: String!, subscriptionItemId: String!, updateMode: String, subscriptionReasonInput: SubscriptionReasonInput): Subscription

File: __mocks__/stories/subscriptionCollectionMock.ts [hunk 3]
```diff
@@ -335,3 +335,12 @@ export const noSubscriptionMock: { subscriptions: SubscriptionCollection } = {
     items: [],
   },
 }
+
+export const deleteSubscriptionMock: { subscription: SubscriptionReason } = {
```
Comment (Kibo-TriptiV) [inline __mocks__/stories/subscriptionCollectionMock.ts]: Return type of delete subscription is Subscription, should we move it to SubscriptionMock instead SubscriptionCollectionMock file?

File: hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @module useSkipNextSubscriptionMutation
+ */
+import { useMutation, useQueryClient } from 'react-query'
+
+import { makeGraphQLClient } from '@/lib/gql/client'
+import { deleteSubscriptionMutation } from '@/lib/gql/mutations'
+import { subscriptionKeys } from '@/lib/react-query/queryKeys'
+
+import { SubscriptionReasonInput } from '@/lib/gql/types'
```
Comment (Kibo-TriptiV) [inline hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: use type

**Review by Kibo-TriptiV — CHANGES_REQUESTED** [review]
Please take a look.

File: components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx (hunk 1 above)
Comment (kibo-Kripa) [reply components/my-account/Subscription/SubscriptionItem/SubscriptionItem.tsx]: Done

File: hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts (hunk 2 above)
Comment (kibo-Kripa) [reply hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: Done

File: __mocks__/stories/subscriptionCollectionMock.ts (hunk 3 above)
Comment (kibo-Kripa) [reply __mocks__/stories/subscriptionCollectionMock.ts]: Done

File: hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts (hunk 4 above)
Comment (kibo-Kripa) [reply hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: done

File: hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/**
+ * @module useSkipNextSubscriptionMutation
```
Comment (kibo-sushant) [inline hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: is this name correct?

Comment (kibo-Kripa) [reply hooks/mutations/useSubscription/useDeleteSubscriptionMutation/useDeleteSubscriptionMutation.ts]: Corrected

File: lib/helpers/buildSubscriptionParams.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export const buildSubscriptionParams = (subscriptionId: string, subscriptionItemId: string) => {
```
Comment (Kibo-TriptiV) [inline lib/helpers/buildSubscriptionParams.ts]: Not changes as per comment on Pause PR.

File: __mocks__/stories/subscriptionMock.ts [hunk 7]
```diff
@@ -41,3 +41,12 @@ export const subscriptionMock = {
     nextOrderDate: 1683072000000,
   },
 }
+
+export const deleteSubscriptionMock: { subscription: SubscriptionReason } = {
```
Comment (Kibo-TriptiV) [inline __mocks__/stories/subscriptionMock.ts]: Not changes as per comment on Pause PR.

**Review by Kibo-TriptiV — CHANGES_REQUESTED** [review]
PauseSubscription and DeleteSubscription PRs are almost similar, please do changes according to both review comments, if missing anything.

File: lib/helpers/buildSubscriptionParams.ts (hunk 6 above)
Comment (kibo-Kripa) [reply lib/helpers/buildSubscriptionParams.ts]: Sorry I missed that! Done now

File: __mocks__/stories/subscriptionMock.ts (hunk 7 above)
Comment (kibo-Kripa) [reply __mocks__/stories/subscriptionMock.ts]: done

**Review by kibo-sushant — APPROVED** [review]

**Review by Kibo-TriptiV — APPROVED** [review]

**Pull request merged**